lib/core/util.js

```diff
@@ -26,6 +26,59 @@ function isBlobLike (object) {
   )
 }
 
+function isDate (val) {
+  return toString.call(val) === '[object Date]'
+}
+
+function isObject (val) {
+  return val !== null && typeof val === 'object'
+}
+
+// this escapes all non-uri friendly characters, but restores : which is commonly used for date formatting
+function encode (val) {
+  return encodeURIComponent(val).replace(/%3A/gi, ':')
+}
+
+// based on https://github.com/axios/axios/blob/63e559fa609c40a0a460ae5d5a18c3470ffc6c9e/lib/helpers/buildURL.js (MIT license)
+function buildURL (url, queryParams) {
+  if (url.includes('?')) {
+    throw new Error('Query params cannot be passed when url contains "?" already.')
+  }
+
+  const parts = []
+  for (let [key, val] of Object.entries(queryParams)) {
+    if (val === null || typeof val === 'undefined') {
+      continue
+    }
+
+    if (!Array.isArray(val)) {
+      val = [val]
+    }
+
+    for (let v of val) {
+      if (isDate(v)) {
+        v = v.toISOString()
```

Comment:
I guess it's the same a `JSON.stringify` does so maybe not too magical.

Reply:
It is more on opinionated side, I agree. Should we throw an error, same way as with Objects, and let users format themselves?

Reply:
I'd prefer throwing.

Reply:
Just to be clear I'm happy to revisit these throwing cases later. But in order to land this PR sooner rather than later I think throwing is appropriate.